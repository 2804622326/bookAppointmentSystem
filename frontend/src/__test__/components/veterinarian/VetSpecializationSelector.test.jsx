import React from 'react';
import { render, waitFor, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import VetSpecializationSelector from '../../../components/veterinarian/VetSpecializationSelector.jsx';

jest.mock('../../../components/veterinarian/VeterinarianService.js', () => ({
  getAllSpecializations: jest.fn(() => Promise.resolve({ data: ['Surg'] }))
}));

test('loads specializations and handles change', async () => {
  const onChange = jest.fn();
  const { asFragment } = render(<VetSpecializationSelector value='' onChange={onChange} />);
  await waitFor(() => screen.getByText('Surg'));
  fireEvent.change(screen.getByRole('combobox'), { target: { value: 'Surg' } });
  expect(onChange).toHaveBeenCalled();
  expect(asFragment()).toMatchSnapshot();
});
