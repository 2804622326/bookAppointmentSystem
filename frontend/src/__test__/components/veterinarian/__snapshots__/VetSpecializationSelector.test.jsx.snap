// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads specializations and handles change 1`] = `
<DocumentFragment>
  <div
    class="mb-4 col"
  >
    <select
      class="form-control"
      id="specialization"
      name="specialization"
      required=""
    >
      <option
        value=""
      >
        ...select specialization...
      </option>
      <option
        value="Surg"
      >
        Surg
      </option>
      <option
        value="Add-New"
      >
        Add New
      </option>
    </select>
  </div>
</DocumentFragment>
`;
